src/codemodder/codemods/https_connection.py

```diff
@@ -0,0 +1,125 @@
+from typing import Sequence
+from libcst import matchers
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.metadata import (
+    PositionProvider,
+)
+from codemodder.codemods.base_codemod import (
+    BaseCodemod,
+    CodemodMetadata,
+    ReviewGuidance,
+)
+from codemodder.codemods.change import Change
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from codemodder.file_context import FileContext
+import libcst as cst
+from libcst.codemod import (
+    Codemod,
+    CodemodContext,
+    VisitorBasedCodemodCommand,
+)
+
+
+class HTTPSConnection(BaseCodemod, Codemod):
+    METADATA = CodemodMetadata(
+        DESCRIPTION=("Enforce HTTPS connection"),
+        NAME="https-connection",
+        REVIEW_GUIDANCE=ReviewGuidance.MERGE_WITHOUT_REVIEW,
+    )
+    CHANGE_DESCRIPTION = "Enforced HTTPS connection"
+    SUMMARY = "Changes HTTPConnectionPool to HTTPSConnectionPool to enforce secure connection."
+
+    METADATA_DEPENDENCIES = (PositionProvider,)
+
+    matching_functions = {
+        "urllib3.HTTPConnectionPool",
+        "urllib3.connectionpool.HTTPConnectionPool",
+    }
+
+    def __init__(self, codemod_context: CodemodContext, file_context: FileContext):
+        Codemod.__init__(self, codemod_context)
+        BaseCodemod.__init__(self, file_context)
+        self.line_exclude = file_context.line_exclude
+        self.line_include = file_context.line_include
+
+    def transform_module_impl(self, tree: cst.Module) -> cst.Module:
+        visitor = ConnectionPollVisitor(self.context, self.file_context)
+        result_tree = visitor.transform_module(tree)
+        self.file_context.codemod_changes.extend(visitor.changes_in_file)
+        return result_tree
+
+
+class ConnectionPollVisitor(VisitorBasedCodemodCommand, NameResolutionMixin):
+    METADATA_DEPENDENCIES = (PositionProvider,)
+
+    def __init__(self, codemod_context: CodemodContext, file_context: FileContext):
+        super().__init__(codemod_context)
+        self.line_exclude = file_context.line_exclude
+        self.line_include = file_context.line_include
+        self.changes_in_file: list[Change] = []
+
+    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
+        pos_to_match = self.node_position(original_node)
+        line_number = pos_to_match.start.line
+        if self.filter_by_path_includes_or_excludes(pos_to_match):
+            true_name = self.find_base_name(original_node.func)
+            if (
+                self._is_direct_call_from_imported_module(original_node)
```

Comment:
It would be cool to generalize this with `filter_by_result` somehow.

Reply:
That is on my (personal) backlog. But it is not happening before we start cleaning up the overall API (specifically our `__init__` calls).